python/unitytrainers/ppo/trainer.py

```diff
@@ -231,31 +260,41 @@ def add_experiences(self, curr_all_info: AllBrainInfo, next_all_info: AllBrainIn
                 idx = stored_info.agents.index(agent_id)
                 next_idx = next_info.agents.index(agent_id)
                 if not stored_info.local_done[idx]:
-                    if self.use_observations:
+                    if self.use_visual_obs:
                         for i, _ in enumerate(stored_info.visual_observations):
-                            self.training_buffer[agent_id]['observations%d' % i].append(stored_info.visual_observations[i][idx])
-                    if self.use_states:
-                        self.training_buffer[agent_id]['states'].append(stored_info.vector_observations[idx])
+                            self.training_buffer[agent_id]['visual_obs%d' % i].append(stored_info.visual_observations[i][idx])
+                            self.training_buffer[agent_id]['next_visual_obs%d' % i].append(next_info.visual_observations[i][idx])
+                    if self.use_vector_obs:
+                        self.training_buffer[agent_id]['vector_obs'].append(stored_info.vector_observations[idx])
+                        self.training_buffer[agent_id]['next_vector_obs'].append(next_info.vector_observations[next_idx])
                     if self.use_recurrent:
                         if stored_info.memories.shape[1] == 0:
                             stored_info.memories = np.zeros((len(stored_info.agents), self.m_size))
                         self.training_buffer[agent_id]['memory'].append(stored_info.memories[idx])
                     actions = stored_take_action_outputs[self.model.output]
                     if self.is_continuous_action:
                         actions_pre = stored_take_action_outputs[self.model.output_pre]
+                        self.training_buffer[agent_id]['actions_pre'].append(actions_pre[idx])
                     a_dist = stored_take_action_outputs[self.model.all_probs]
                     value = stored_take_action_outputs[self.model.value]
                     self.training_buffer[agent_id]['actions'].append(actions[idx])
-                    if self.is_continuous_action:
-                        self.training_buffer[agent_id]['actions_pre'].append(actions_pre[idx])
                     self.training_buffer[agent_id]['prev_action'].append(stored_info.previous_vector_actions[idx])
                     self.training_buffer[agent_id]['masks'].append(1.0)
-                    self.training_buffer[agent_id]['rewards'].append(next_info.rewards[next_idx])
+                    if self.use_curiosity:
```

Comment:
Would it make more sense to always have an intrinsic reward but only have it non-zero when curiosity is used ?

Reply:
If we did that we'd need some code to generate zeroes and keep track of them, which doesn't seem to be as elegant.